packages/components/src/tooltip/test/index.tsx

```diff
@@ -15,321 +15,408 @@ import { shortcutAriaLabel } from '@wordpress/keycodes';
+					<button>Click me</button>
+				</>
+			);
 
-		await user.hover( screen.getByRole( 'button', { name: /Button/i } ) );
+			const anchor = screen.getByRole( 'button', {
+				name: 'Tooltip anchor',
+			} );
 
-		// Advance time by default delay
-		await new Promise( ( resolve ) =>
-			setTimeout( resolve, TOOLTIP_DELAY )
-		);
+			expect( anchor ).toBeVisible();
 
-		// Tooltip hasn't appeared yet
-		expect(
-			screen.queryByRole( 'tooltip', { name: /tooltip text/i } )
-		).not.toBeInTheDocument();
+			// Hover over the anchor, tooltip should show
+			await hover( anchor );
+			await waitForTooltipToShow();
 
-		// wait for additional delay for tooltip to appear
-		await waitFor(
-			() =>
-				new Promise( ( resolve ) =>
-					setTimeout( resolve, ADDITIONAL_DELAY )
-				)
-		);
+			// Click the anchor, tooltip should not hide
+			await click( anchor );
+			await waitForTooltipToShow();
 
-		expect(
-			screen.getByRole( 'tooltip', { name: /tooltip text/i } )
-		).toBeVisible();
-
-		await cleanupTooltip( user );
+			// Click another button, tooltip should hide
+			await click( screen.getByRole( 'button', { name: 'Click me' } ) );
+			await waitForTooltipToHide();
+		} );
 	} );
 
-	it( 'should show tooltip when an element is disabled', async () => {
-		const user = userEvent.setup();
+	describe( 'delay', () => {
+		it( 'should respect custom delay prop when showing tooltip', async () => {
+			const ADDITIONAL_DELAY = 100;
 
-		render(
-			<Tooltip { ...props }>
-				<Button aria-disabled>Button</Button>
-			</Tooltip>
-		);
+			render(
+				<Tooltip
+					{ ...props }
+					delay={ TOOLTIP_DELAY + ADDITIONAL_DELAY }
+				/>
+			);
 
-		const button = screen.getByRole( 'button', { name: /Button/i } );
+			const anchor = screen.getByRole( 'button', {
+				name: 'Tooltip anchor',
+			} );
+			expect( anchor ).toBeVisible();
 
-		expect( button ).toBeVisible();
-		expect( button ).toHaveAttribute( 'aria-disabled' );
+			// Hover over the anchor
+			await hover( anchor );
+			expectTooltipToBeHidden();
 
-		await user.hover( button );
+			// Advance time by default delay
+			await sleep( TOOLTIP_DELAY );
 
-		expect(
-			await screen.findByRole( 'tooltip', { name: /tooltip text/i } )
-		).toBeVisible();
+			// Tooltip hasn't appeared yet
+			expectTooltipToBeHidden();
 
-		await cleanupTooltip( user );
-	} );
+			// Wait for additional delay for tooltip to appear
+			await sleep( ADDITIONAL_DELAY );
+			await waitForTooltipToShow();
 
-	it( 'should not show tooltip if the mouse leaves the tooltip anchor before set delay', async () => {
-		const user = userEvent.setup();
-		const onMouseEnterMock = jest.fn();
-		const onMouseLeaveMock = jest.fn();
-		const MOUSE_LEAVE_DELAY = TOOLTIP_DELAY - 200;
+			// Hover outside of the anchor, tooltip should hide
+			await hoverOutside();
+			await waitForTooltipToHide();
+		} );
 
-		render(
-			<>
+		it( 'should not show tooltip if the mouse leaves the tooltip anchor before set delay', async () => {
```

Comment:
@diegohaz this test keeps failing on my local machine when I run the whole suite, but passes then run in isolation.

It looks like there is still some "leak" across tests even after using `@ariakit/test`, but I can't understand what it is — do you have any ideas?

Reply:
Tooltips appear instantly if another tooltip has just been hidden. This is controlled by a global state, which is probably shared among multiple tests within the same file. These tests don't exactly run in total isolation.

If this is the issue, you might need to introduce a delay at the start to ensure tooltips from other tests aren't interfering with this one.

Reply:
> Tooltips appear instantly if another tooltip has just been hidden. This is controlled by a global state, which is probably shared among multiple tests within the same file. These tests don't exactly run in total isolation.

That seems to be the issue, since the test passes if I set the `skipTimeout: 0` (although I'd prefer to leave the `skipTimeout` delay around if possible).

> If this is the issue, you might need to introduce a delay at the start to ensure tooltips from other tests aren't interfering with this one.

I tried adding a delay specifically for that test, but I keep getting failures. The only way to reliably have all tests pass seems to be to add

```
	afterEach( async () => {
		await sleep( 300 );
	} );
```

although I'd prefer to avoid slowing down the whole suite if possible. Do you have any suggestions?

Reply:
For now, I'll have to resort to waiting those extra `300ms` at the end of each test to make sure that the "skip timeout" behaviour is not triggered.

Reply:
FYI, this delay might not be required anymore with `@ariakit/react@0.4.2`.

Reply:
PR to remove it: https://github.com/WordPress/gutenberg/pull/60897